framework/curl_client.py

```diff
@@ -0,0 +1,310 @@
+        """Received statuses counters, like:
+        {200: 1, 400: 2}
+        """
+        return dict(self._statuses)
+
+    @statuses.setter
+    def statuses(self, value):
+        # ignore attribute initialization by `Client`
+        pass
+
+    @property
+    def last_stats(self) -> Dict[str, Any]:
+        """Information about last completed transfer.
+        See https://curl.se/docs/manpage.html#-w
+        for the list of available variables.
+        """
+        return (self._stats or [None])[-1]
+
+    @property
+    def stats(self) -> List[Dict[int, int]]:
+        """List of stats of all transfers"""
+        return list(self._stats)
+
+    @property
+    def binary_version(self) -> Optional[str]:
+        """curl binary version, parsed from the latest transfer."""
+        if self.last_stats:
+            return self._parse_binary_version(self.last_stats["curl_version"])
+
+    @property
+    def cookie_jar_path(self):
+        """Path to save/load cookies."""
+        return Path(self.workdir) / "curl-default.jar"
+
+    @property
+    def output_path(self):
+        """Path to write stdout (response)."""
+        return Path(self.workdir) / "curl-output"
+
+    @property
+    def headers_dump_path(self):
+        """Path do dump received headers."""
+        return Path(self.workdir) / "curl-default.hdr"
+
+    def clear_cookies(self):
+        """Delete cookies from previous runs."""
+        self.cookie_jar_path.unlink(missing_ok=True)
+
+    def form_command(self):
+        options = ["--no-progress-meter", "--create-dirs"]
+
+        if self.dump_headers:
+            options.append(f"--dump-header '{self.headers_dump_path}'")
+
+        if self.disable_output:
+            options.append("--silent --show-error --output /dev/null")
+        else:
+            options.append(f"--write-out '%{{json}}{self._output_delimeter}'")
+            options.append(f"--output '{self.output_path}'")
+
+        if self.save_cookies:
+            options.append(f"--cookie-jar '{self.cookie_jar_path}'")
+        if self.load_cookies:
+            options.append(f"--cookie '{self.cookie_jar_path}'")
+
+        for header, value in (self.headers or {}).items():
+            options.append(f"--header '{header}: {value}'")
+
+        if self.data:
+            options.append(f"--data '{self.data}'")
+
+        if self.http2:
+            options.append("--http2-prior-knowledge")
+        else:
+            options.append("--http1.1")
+
+        if self.ssl and self.insecure:
+            options.append("--insecure")
+
+        if self.parallel:
+            options.append("--parallel")
+            options.append("--parallel-immediate")
+            options.append(f"--parallel-max {self.parallel}")
+
+        cmd = " ".join([self.bin] + options + self.options + [f"'{self.uri}'"])
+        tf_cfg.dbg(3, f"Curl command formatted: {cmd}")
+        return cmd
+
+    def parse_out(self, stdout, stderr):
+        self.requests += 1
```

Comment:
It should be `self.request = 1` because it's confusing. `start()` method clears stats and we always get value 1. And also it does not show the number of requests for concurrent connection.

Reply:
Fixed, `self.requests` now is a calculated field based on number of received stats (if stats was parsed) or responses.
Clearing of `responses`, `stats` and `statuses` are fixed too.